src/inmanta/data/model.py

```diff
@@ -763,28 +765,40 @@ class DiscoveredResource(BaseModel):
     :param values: The actual resource
     :param managed_resource_uri: URI of the resource with the same ID that is already
         managed by the orchestrator e.g. "/api/v2/resource/<rid>". Or None if the resource is not managed.
-    :param discovery_resource_id: Resource id of the (managed) discovery resource that reported this
-        discovered resource.
     """
 
-    discovered_resource_id: ResourceIdStr
+    discovered_resource_id: ResourceId
     values: dict[str, object]
     managed_resource_uri: Optional[str] = None
 
-    # This field is excluded during serialization in favour of the discovery_resource_uri
-    discovery_resource_id: Optional[ResourceIdStr] = Field(default=None, exclude=True)
-
     _validate_discovered_rid = field_validator("discovered_resource_id")(is_resource_id)
-    _validate_discovery_rid = field_validator("discovery_resource_id")(functools.partial(is_resource_id, allow_none=True))
+
+    def to_dao(self, env: uuid.UUID) -> "data.DiscoveredResource":
+        return data.DiscoveredResource(
+            discovered_resource_id=self.discovered_resource_id,
+            values=self.values,
+            discovered_at=datetime.datetime.now(),
+            environment=env,
+        )
+
+
+class LinkedDiscoveredResource(DiscoveredResource):
```

Comment:
Perhaps add a comment (not docstring because that's included in the docs) as to why we need this class and what its purpose is (i.e. API input).